zyte_test_websites/ecommerce/extraction.py

```diff
@@ -0,0 +1,181 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, cast
+from urllib.parse import urljoin
+
+import attrs
+from web_poet import field
+from zyte_common_items import (
+    AdditionalProperty,
+    AggregateRating,
+    Breadcrumb,
+    Link,
+    ProbabilityRequest,
+    ProductFromList,
+    ProductFromListSelectorExtractor,
+    ProductListPage,
+    ProductNavigationPage,
+    ProductPage,
+    Request,
+)
+from zyte_common_items.pages import PriceMixin
+
+if TYPE_CHECKING:
+    from parsel import Selector, SelectorList
+
+__all__ = [
+    "TestProductListPage",
+    "TestProductNavigationPage",
+    "TestProductPage",
+]
+
+
+class TestProductPage(ProductPage):
+    @field
```

Comment:
```suggestion
    @field(cached=True)
```
as this one is used by other fields